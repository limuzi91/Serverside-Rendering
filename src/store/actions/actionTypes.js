export const FETCH_JS_START = "FETCH_JS_START";
export const FETCH_JS_SUCCESS = "FETCH_JS_SUCCESS";
export const FETCH_JS_FAIL = "FETCH_JS_FAIL";

export const FETCH_JAVA_START = "FETCH_JAVA_START";
export const FETCH_JAVA_SUCCESS = "FETCH_JAVA_SUCCESS";
export const FETCH_JAVA_FAIL = "FETCH_JAVA_FAIL";

export const SET_TEXT_FILTER = "SET_TEXT_FILTER";
export const RESET_FILTERS = "RESET_FILTERS";

export const SAVE_COMMENT = "SAVE_COMMENT";
export const FETCH_COMMENTS = "FETCH_COMMENTS";
