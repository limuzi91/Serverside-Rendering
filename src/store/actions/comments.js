import axios from "axios";
import { SAVE_COMMENT, FETCH_COMMENTS } from "./actionTypes";

export const saveComment = comment => {
  return {
    type: SAVE_COMMENT,
    payload: comment
  };
};

export const fetchCommentsSuccess = comments => ({
  type: FETCH_COMMENTS,
  payload: comments
});

export const fetchComments = () => {
  return dispatch => {
    axios.get("http://jsonplaceholder.typicode.com/comments").then(response => {
      //console.log(response.data.slice(0, 50));
      dispatch(fetchCommentsSuccess(response.data.slice(0, 50)));
    });
  };
};
