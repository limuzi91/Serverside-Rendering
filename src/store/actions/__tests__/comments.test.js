import { saveComment, fetchCommentsSuccess, fetchComments } from "../comments";
import { SAVE_COMMENT } from "../actionTypes";
import moxios from "moxios";

// Below are two approachs to test action creators

describe("test saveComment", () => {
  it("has the correct type", () => {
    const action = saveComment();

    expect(action.type).toEqual(SAVE_COMMENT);
  });

  it("has the correct payload", () => {
    const action = saveComment("New Comment");

    expect(action.payload).toEqual("New Comment");
  });
});

it("test fetchCommentsSuccess through the Snapshot approach", () => {
  expect(fetchCommentsSuccess(["comment 1", "comment 2"])).toMatchSnapshot();
});

it("test thunk (fetchComments) after testing action creator(fetchCommentsSuccess)", done => {
  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    const thunk = fetchComments();
    thunk(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: ["Fetched #1", "Fetched #2"]
        })
        .then(() => {
          expect(request.url).toEqual(
            `http://jsonplaceholder.typicode.com/comments`
          );
          expect(dispatchMock).toBeCalledWith(
            fetchCommentsSuccess(["Fetched #1", "Fetched #2"])
          );
          done();
        });
    });
  });
});
