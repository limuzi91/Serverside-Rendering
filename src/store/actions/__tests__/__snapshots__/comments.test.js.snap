// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test fetchCommentsSuccess through the Snapshot approach 1`] = `
Object {
  "payload": Array [
    "comment 1",
    "comment 2",
  ],
  "type": "FETCH_COMMENTS",
}
`;
